appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java

```diff
@@ -119,36 +121,37 @@
     private ProcessType processType;
 
     // Keep track of contexts for managed bean instances instantiated via JCDI
-    private Map<BundleDescriptor, Map<Object, JCDIService.JCDIInjectionContext>> jcdiManagedBeanInstanceMap =
-            new HashMap<BundleDescriptor, Map<Object, JCDIService.JCDIInjectionContext>>();
+    private Map<BundleDescriptor, Map<Object, JCDIService.JCDIInjectionContext<?>>> jcdiManagedBeanInstanceMap = new HashMap<>();
 
     // Used to hold managed beans in app client container
-    private Map<String, NamingObjectProxy> appClientManagedBeans = new HashMap<String, NamingObjectProxy>();
+    private Map<String, NamingObjectProxy> appClientManagedBeans = new HashMap<>();
 
+    @Override
     public void postConstruct() {
         events.register(this);
         processType = processEnv.getProcessType();
     }
 
-    public void event(Event event) {
+    @Override
+    public void event(Event<?> event) {
 
          if (event.is(Deployment.APPLICATION_LOADED) ) {
-             ApplicationInfo info =  Deployment.APPLICATION_LOADED.getHook(event);
+             ApplicationInfo info =  Deployment.APPLICATION_LOADED.getHook((Event<ApplicationInfo>) event);
```

Comment:
There is no else since this is a general event listener mechanism so you will get all kinds of events even if you are only interested in some. Implementations then just pick out those they originally wanted to be connected to. This is a shortcoming of the current `EventListener` mechanism.